src/main/webapp/js/countryCodes.js

```diff
@@ -0,0 +1,532 @@
+    ["KN","KNA",659,"ISO 3166-2:KN"],
+    ["LC","LCA",662,"ISO 3166-2:LC"],
+    ["MF","MAF",663,"ISO 3166-2:MF"],
+    ["PM","SPM",666,"ISO 3166-2:PM"],
+    ["VC","VCT",670,"ISO 3166-2:VC"],
+    ["WS","WSM",882,"ISO 3166-2:WS"],
+    ["SM","SMR",674,"ISO 3166-2:SM"],
+    ["ST","STP",678,"ISO 3166-2:ST"],
+    ["SA","SAU",682,"ISO 3166-2:SA"],
+    ["SN","SEN",686,"ISO 3166-2:SN"],
+    ["RS","SRB",688,"ISO 3166-2:RS"],
+    ["SC","SYC",690,"ISO 3166-2:SC"],
+    ["SL","SLE",694,"ISO 3166-2:SL"],
+    ["SG","SGP",702,"ISO 3166-2:SG"],
+    ["SX","SXM",534,"ISO 3166-2:SX"],
+    ["SK","SVK",703,"ISO 3166-2:SK"],
+    ["SI","SVN",705,"ISO 3166-2:SI"],
+    ["SB","SLB",90,"ISO 3166-2:SB"],
+    ["SO","SOM",706,"ISO 3166-2:SO"],
+    ["ZA","ZAF",710,"ISO 3166-2:ZA"],
+    ["GS","SGS",239,"ISO 3166-2:GS"],
+    ["SS","SSD",728,"ISO 3166-2:SS"],
+    ["ES","ESP",724,"ISO 3166-2:ES"],
+    ["LK","LKA",144,"ISO 3166-2:LK"],
+    ["SD","SDN",729,"ISO 3166-2:SD"],
+    ["SR","SUR",740,"ISO 3166-2:SR"],
+    ["SJ","SJM",744,"ISO 3166-2:SJ"],
+    ["SZ","SWZ",748,"ISO 3166-2:SZ"],
+    ["SE","SWE",752,"ISO 3166-2:SE"],
+    ["CH","CHE",756,"ISO 3166-2:CH"],
+    ["SY","SYR",760,"ISO 3166-2:SY"],
+    ["TW","TWN",158,"ISO 3166-2:TW"],
+    ["TJ","TJK",762,"ISO 3166-2:TJ"],
+    ["TZ","TZA",834,"ISO 3166-2:TZ"],
+    ["TH","THA",764,"ISO 3166-2:TH"],
+    ["TL","TLS",626,"ISO 3166-2:TL"],
+    ["TG","TGO",768,"ISO 3166-2:TG"],
+    ["TK","TKL",772,"ISO 3166-2:TK"],
+    ["TO","TON",776,"ISO 3166-2:TO"],
+    ["TT","TTO",780,"ISO 3166-2:TT"],
+    ["TN","TUN",788,"ISO 3166-2:TN"],
+    ["TR","TUR",792,"ISO 3166-2:TR"],
+    ["TM","TKM",795,"ISO 3166-2:TM"],
+    ["TC","TCA",796,"ISO 3166-2:TC"],
+    ["TV","TUV",798,"ISO 3166-2:TV"],
+    ["UG","UGA",800,"ISO 3166-2:UG"],
+    ["UA","UKR",804,"ISO 3166-2:UA"],
+    ["AE","ARE",784,"ISO 3166-2:AE"],
+    ["GB","GBR",826,"ISO 3166-2:GB"],
+    ["US","USA",840,"ISO 3166-2:US"],
+    ["UM","UMI",581,"ISO 3166-2:UM"],
+    ["UY","URY",858,"ISO 3166-2:UY"],
+    ["UZ","UZB",860,"ISO 3166-2:UZ"],
+    ["VU","VUT",548,"ISO 3166-2:VU"],
+    ["VE","VEN",862,"ISO 3166-2:VE"],
+    ["VN","VNM",704,"ISO 3166-2:VN"],
+    ["VG","VGB",92,"ISO 3166-2:VG"],
+    ["VI","VIR",850,"ISO 3166-2:VI"],
+    ["WF","WLF",876,"ISO 3166-2:WF"],
+    ["EH","ESH",732,"ISO 3166-2:EH"],
+    ["YE","YEM",887,"ISO 3166-2:YE"],
+    ["ZM","ZMB",894,"ISO 3166-2:ZM"],
+    ["ZW","ZWE",716,"ISO 3166-2:ZW"]
+];
+
+var isoCountriesAlpha3 = {};
+
+// Mapping from alpha 2 to alpha 3
+for (var i = 0; i < codes.length; i++) {
+    if (isoCountries.hasOwnProperty(codes[i][0])) {
+        // Mapping found
+        isoCountriesAlpha3[codes[i][1]] = isoCountries[codes[i][0]];
+    }
+}
+
+var countryToCode = {};
+
+// Generate mapping from name to alpha 2
+for (var countryCode in isoCountriesAlpha3) {
+    if (isoCountriesAlpha3.hasOwnProperty(countryCode)) {
+        countryToCode[isoCountriesAlpha3[countryCode]] = countryCode;
+    }
+}
```

Comment:
I realize this might be necessary because the data comes from two different sources. Can you do some processing and combine them to make one centralized data source? That way you can also cross-check if there are unpaired `code` and `isoCountry`.
One good way I can think of to do it fast is converting the data source to JSON and then downloading it.